java/java.graph/src/org/netbeans/modules/java/graph/FruchtermanReingoldLayout.java

```diff
@@ -382,7 +382,8 @@ private void relayoutNonFixed(NodeWidget w) {
         r = 30;
         theta = 0;
         w.setFixed(false);
-        while (true) {
+        // 48 - ~3 times round?
+        for (int i = 0; i < 48; i++) {
```

Comment:
If I read this right it will stop moving the node if it can't find a free spot after a few attempts. Makes sense.

Reply:
Yes. What I'm less sure about is why it doesn't come to an end given the increase in `r`. Increasing the increment of `r` does get the loop to end on my test project (with a result similar to that of increasing the forces), but takes longer, and is potentially still liable to get stuck with a different set of nodes.

This whole thing could do with a proper look after NB25.  I find the horizontal tree (in popup menu -> layout) a potentially more useful display for smaller projects, and might be a better default given it uses a lot less CPU.